Lib/zipfile.py

```diff
@@ -341,6 +341,8 @@ class ZipInfo (object):
     )
 
     def __init__(self, filename="NoName", date_time=(1980,1,1,0,0,0)):
+        if isinstance(filename, os.PathLike):
+            filename = os.fspath(filename)
```

Comment:
This should be just `filename = os.fspath(filename)`, the `if ...:` is not needed.

Reply:
You're right. I assumed it was the right way because there's a similar check in ZipInfo.from_file, but that one is needed to accept fileno as integers, my bad.

Anyway thanks. I've pushed the suggested change.